app/ts/simulation/services/EtherScanAbiFetcher.ts

```diff
@@ -20,20 +22,32 @@ export function isValidAbi(abi: string) {
 	}
 }
 
-export async function fetchAbiFromEtherscan(contractAddress: EthereumAddress) {
-	const json = await fetchJson(`https://api.etherscan.io/api?module=contract&action=getsourcecode&address=${ addressString(contractAddress) }&apiKey=${ EtherscanABIKey }`)
+export function getBlockExplorer(chainId: ChainIdWithUniversal, rpcEntries: RpcEntries) {
+	if (chainId === 'AllChains') return undefined
+	const primaryRpc = rpcEntries.find((rpc) => rpc.chainId === chainId && rpc.primary)
+	if (primaryRpc !== undefined && 'blockExplorer' in primaryRpc && primaryRpc.blockExplorer !== undefined) return primaryRpc.blockExplorer
```

Comment:
blockExplorer is required in `primaryRpc` so can skip the `in` check

```suggestion
	if (primaryRpc !== undefined && primaryRpc.blockExplorer !== undefined) return primaryRpc.blockExplorer
```